sewer/tests/test_Client.py

```diff
@@ -359,6 +337,126 @@ def setUp(self):
 
 
+class TestClientDnsApiCompatibility(TestCase):
+    """
+    Test Acme client for the existing dns provider api.
+    """
+
+    def setUp(self):
+        self.domain_name = "example.com"
+        with mock.patch("requests.post") as mock_requests_post, mock.patch(
+            "requests.get"
+        ) as mock_requests_get:
+            mock_requests_post.return_value = test_utils.MockResponse()
+            mock_requests_get.return_value = test_utils.MockResponse()
+
+            self.dns_class = test_utils.ExmpleDnsProvider()
+            self.client = sewer.Client(
+                domain_name=self.domain_name,
+                dns_class=self.dns_class,
+                ACME_REQUEST_TIMEOUT=1,
+                ACME_AUTH_STATUS_WAIT_PERIOD=0,
+                ACME_DIRECTORY_URL=ACME_DIRECTORY_URL_STAGING,
+            )
+
+    def test_get_get_acme_endpoints_failure_results_in_exception_with(self):
+        with mock.patch("requests.post") as mock_requests_post, mock.patch(
+            "requests.get"
+        ) as mock_requests_get:
+            mock_requests_post.return_value = test_utils.MockResponse(status_code=409)
+            mock_requests_get.return_value = test_utils.MockResponse(status_code=409)
+
+            def mock_create_acme_client():
+                sewer.Client(
+                    domain_name="example.com",
+                    dns_class=test_utils.ExmpleDnsProvider(),  # NOTE: dns_class used here
+                    ACME_DIRECTORY_URL=ACME_DIRECTORY_URL_STAGING,
+                )
+
+            self.assertRaises(ValueError, mock_create_acme_client)
+            with self.assertRaises(ValueError) as raised_exception:
+                mock_create_acme_client()
+            self.assertIn("Error while getting Acme endpoints", str(raised_exception.exception))
+
+    def test_create_dns_record_is_called(self):
+        with mock.patch("requests.post") as mock_requests_post, mock.patch(
+            "requests.get"
+        ) as mock_requests_get, mock.patch(
+            "sewer.tests.test_utils.ExmpleDnsProvider.create_dns_record"
+        ) as mock_create_dns_record:
+            mock_requests_post.return_value = test_utils.MockResponse()
+            mock_requests_get.return_value = test_utils.MockResponse()
+            self.client.cert()
+            self.assertTrue(mock_create_dns_record.called)
+
+    def test_delete_dns_record_is_called(self):
+        with mock.patch("requests.post") as mock_requests_post, mock.patch(
+            "requests.get"
+        ) as mock_requests_get, mock.patch(
+            "sewer.tests.test_utils.ExmpleDnsProvider.delete_dns_record"
+        ) as mock_delete_dns_record:
+            mock_requests_post.return_value = test_utils.MockResponse()
+            mock_requests_get.return_value = test_utils.MockResponse()
+            self.client.cert()
+            self.assertTrue(mock_delete_dns_record.called)
+
+    def test_right_args_to_client(self):
+        def mock_instantiate_client():
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'mock_instantiate_client'](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5008291)

Reply:
@AlecTroemel `mock_instantiate_client` is never used.